components/src/components/molecules/Select/Select.tsx

```diff
@@ -452,9 +568,49 @@ export const Select = React.forwardRef(
 
     const [menuOpen, setMenuOpen] = React.useState(false)
     const isOpen = !disabled && menuOpen
+
+    const showClearButton = queryValue !== '' && isAutocomplete
+
+    // Set the intrinsic size of the input
+    const minInputSize =
+      typeof inputSizeProps === 'number'
+        ? inputSizeProps
+        : typeof inputSizeProps?.min === 'number'
+        ? inputSizeProps.min
+        : 4
+
+    const maxInputSize =
+      typeof inputSizeProps === 'number'
+        ? inputSizeProps
+        : typeof inputSizeProps?.max === 'number'
+        ? inputSizeProps.max
+        : 20
+
+    const inputSize = Math.min(
+      Math.max(minInputSize, queryValue.length),
+      maxInputSize,
+    )
+
+    const [state, toggle] = useTransition({
+      timeout: {
+        enter: 300,
```

Comment:
this should be zero, otherwise causes a delay on enter

Reply:
I took a look at the example code and I find it interesting how you broke away from how they did it. Your implementation is actually more consistent. Sometimes their implementation causes some timing error from mounting which makes the menu appear immediately. I added the preEnter state to give it an extra frame to prevent the menu from jumping.